homeassistant/components/isy994/number.py

```diff
@@ -1,22 +1,54 @@
 """Support for ISY number entities."""
 from __future__ import annotations
 
-from typing import Any
+from typing import Any, cast
 
+from pyisy.constants import COMMAND_FRIENDLY_NAME, ISY_VALUE_UNKNOWN, PROP_ON_LEVEL
 from pyisy.helpers import EventListener, NodeProperty
+from pyisy.nodes import Node
 from pyisy.variables import Variable
 
-from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.components.number import (
+    ATTR_MAX,
+    ATTR_MIN,
+    ATTR_STEP,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_VARIABLES, Platform
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_VARIABLES,
+    PERCENTAGE,
+    Platform,
+)
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
 
-from .const import CONF_VAR_SENSOR_STRING, DEFAULT_VAR_SENSOR_STRING, DOMAIN
+from .const import (
+    CONF_VAR_SENSOR_STRING,
+    DEFAULT_VAR_SENSOR_STRING,
+    DOMAIN,
+    UOM_8_BIT_RANGE,
+)
 from .helpers import convert_isy_value_to_hass
 
 ISY_MAX_SIZE = (2**32) / 2
+ON_RANGE = (1, 255)  # Off is not included
+CONTROL_DESC = {
+    PROP_ON_LEVEL: {
+        ATTR_MIN: 1.0,
+        ATTR_MAX: 100.0,
+        ATTR_STEP: 1.0,
+        ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
+    }
+}
```

Comment:
This looks like a good use case for `NumberEntityDescription`. Not sure if we need it yet but `replace` can also be used with these https://docs.python.org/3/library/dataclasses.html#dataclasses.replace

Reply:
Revised. No need to use replace for these yet, but made use of it to simplify the `_init` description for Variables.